src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs

```diff
@@ -119,6 +127,9 @@ public QueueProcessor(QueueProcessorFactoryContext context)
                 if (message.DequeueCount >= MaxDequeueCount)
                 {
                     await CopyMessageToPoisonQueueAsync(message, _poisonQueue, cancellationToken);
+
+                    message.SetProperties(id, popReceipt, insertionTime, nextVisibleTime, expirationTime);
```

Comment:
Also, can you comment the code with a TEMP that points to an issue in our repo tracking addressing this once we get Clone and move to 8.x (our next major version)?